tests/e2e/util.go

```diff
@@ -2103,12 +2103,12 @@ func createPod(client clientset.Interface, namespace string, nodeSelector map[st
 	// Waiting for pod to be running
 	err = fpod.WaitForPodNameRunningInNamespace(client, pod.Name, namespace)
 	if err != nil {
-		return nil, fmt.Errorf("pod %q is not Running: %v", pod.Name, err)
+		return pod, fmt.Errorf("pod %q is not Running: %v", pod.Name, err)
```

Comment:
Can you add a comment to indicate why we are returning pointer to pod even during failure cases

Reply:
Added a comment  above in "Special notes for your reviewer:" section